extensions/sql/asset/index/src/main/java/org/eclipse/dataspaceconnector/sql/asset/index/SqlAssetIndex.java

```diff
@@ -127,23 +125,53 @@ public Asset deleteById(String assetId) {
         }
     }
 
+    @Override
+    public void accept(AssetEntry item) {
+        Objects.requireNonNull(item);
+
+        accept(item.getAsset(), item.getDataAddress());
+    }
+
     @Override
     public Stream<Asset> queryAssets(AssetSelectorExpression expression) {
         Objects.requireNonNull(expression);
 
-        return getAssetStream(sqlAssetQueries.getSqlAssetListClause());
+        var criteria = expression.getCriteria();
+        var querySpec = QuerySpec.Builder.newInstance().filter(criteria)
+                .offset(0)
+                .limit(Integer.MAX_VALUE) // means effectively no limit
```

Comment:
Would it make sense to have limit in QuerySpec initialised by default with  Integer.MAX_VALUE (effectively no limit)? I see now it's 50

Reply:
i don't think so, because this is really a specific case: we want the `LIMIT ` clause there so as not to complicate the statement generation even more, but effectively there shouldn't be a limit. 
The `QuerySpec` should definitely have sane defaults that don't bork the database :)